include/sparrow/dynamic_bitset.hpp

```diff
@@ -154,6 +178,101 @@ namespace sparrow
         dynamic_bitset_view& operator=(dynamic_bitset_view&&) = default;
     };
 
+    /**
+     *@class bitset_reference
+     */
+    template <class B>
+    class bitset_reference
+    {
+    public:
+
+        using self_type = bitset_reference<B>;
+
+        bitset_reference(const bitset_reference&) = default;
+        bitset_reference(bitset_reference&&) = default;
+
+        self_type& operator=(const self_type&) noexcept;
+        self_type& operator=(self_type&&) noexcept;
+        self_type& operator=(bool) noexcept;
+
+        explicit operator bool() const noexcept;
+
+        bool operator~() const noexcept;
+
+        self_type& operator&=(bool) noexcept;
+        self_type& operator|=(bool) noexcept;
+        self_type& operator^=(bool) noexcept;
+
+    private:
+
+        using block_type = typename B::block_type;
+        using bitset_type = B;
+
+        bitset_reference(bitset_type& bitset, block_type& block, block_type mask);
+
+        void assign(bool) noexcept;
+        void set() noexcept;
+        void reset() noexcept;
+        
+        bitset_type& m_bitset;
+        block_type& m_block;
+        block_type m_mask;
+
+        friend class bitset_iterator<B, false>;
+    };
+
+    /**
+     * @class bitset_iterator
+     */
+    template <class B, bool is_const>
+    class bitset_iterator : public iterator_base
+    <
+        bitset_iterator<B, is_const>,
+        mpl::constify_t<typename B::value_type, is_const>,
+        std::contiguous_iterator_tag,
+        std::conditional_t<is_const, bool, bitset_reference<B>>
+    >
+    {
+    public:
+
+        using block_type = mpl::constify_t<typename B::block_type, is_const>;
+        using bitset_type = mpl::constify_t<B, is_const>;
+        using size_type = typename B::size_type;
+
+        bitset_iterator() noexcept = default;
+        bitset_iterator(bitset_type* bitset, block_type* block, size_type index);
+
+    private:
+
+        using self_type = bitset_iterator<B, is_const>;
+        using base_type = iterator_base
+        <
+            self_type,
+            mpl::constify_t<typename B::value_type, is_const>,
+            std::contiguous_iterator_tag,
+            std::conditional_t<is_const, bool, bitset_reference<B>>
+        >;        
+        using reference = typename base_type::reference;
+        using difference_type = typename base_type::difference_type;
+
+        reference dereference() const;
+        void increment();
+        void decrement();
+        void advance(difference_type n);
+        difference_type distance_to(const self_type& rhs) const;
+        bool equal(const self_type& rhs) const;
+        bool less_than(const self_type& rhs) const;
+
+        bool is_first_bit_of_block(size_type index) const;
+        difference_type distance_to_begin() const;
+
+        bitset_type* p_bitset = nullptr;
+        block_type* p_block = nullptr;
+        size_type m_index;
```

Comment:
Could we specify if `m_index` corresponds to the block-local index or the absolute "logical" index?

For instance for the first case:
```suggestion
        // m_index is block-local index.
        // Invariant: 0 <= m_index < bitset_type::s_bits_per_block
        size_type m_index;
```